napari/layers/base/_base_mouse_bindings.py

```diff
@@ -0,0 +1,171 @@
+    initial_handle_coords_data = generate_transform_box_from_layer(
+        layer, event.dims_displayed
+    )
+    nearby_handle = get_nearby_handle(pos_data, initial_handle_coords_data)
+
+    if nearby_handle is None:
+        return
+
+    # now that we have the nearby handles, other calculations need
+    # the world space handle positions
+    initial_handle_coords = data_to_world(initial_handle_coords_data)
+
+    # initial layer transform so we can calculate changes later
+    initial_affine = layer.affine.set_slice(event.dims_displayed)
+    initial_data2physical = layer._transforms['data2physical'].set_slice(
+        event.dims_displayed
+    )
+    initial_position = pos
+
+    # needed for resize and rotate
+    center = np.mean(
+        initial_handle_coords[
+            [
+                InteractionBoxHandle.TOP_LEFT,
+                InteractionBoxHandle.BOTTOM_RIGHT,
+            ]
+        ],
+        axis=0,
+    )
+
+    yield
+
+    while event.type == 'mouse_move':
+        pos = np.array(event.position)[event.dims_displayed]
+
+        if nearby_handle == InteractionBoxHandle.INSIDE:
+            offset = pos - initial_position
+            new_affine = Affine(translate=offset).compose(initial_affine)
+            layer.affine = layer.affine.replace_slice(
+                event.dims_displayed, new_affine
+            )
+            yield
+        elif nearby_handle == InteractionBoxHandle.ROTATION:
+
+            # calculate the angle between the center-handle vector and the center-mouse vector
+            center_to_handle = (
+                initial_handle_coords[InteractionBoxHandle.ROTATION] - center
+            )
+            center_to_handle /= np.linalg.norm(center_to_handle)
+            center_to_mouse = pos - center
+            center_to_mouse /= np.linalg.norm(center_to_mouse)
+            angle = np.arctan2(
+                center_to_mouse[1], center_to_mouse[0]
+            ) - np.arctan2(center_to_handle[1], center_to_handle[0])
+
+            new_affine = (
+                Affine(translate=center)
+                .compose(Affine(rotate=np.rad2deg(angle)))
+                .compose(Affine(translate=-center))
+                .compose(initial_affine)
+            )
+            layer.affine = layer.affine.replace_slice(
+                event.dims_displayed, new_affine
+            )
+            yield
+        else:
+            # scale transform
+
+            if 'Shift' in event.modifiers:
+                if nearby_handle not in InteractionBoxHandle.corners():
+                    raise ValueError(
+                        'aspect ratio can only be blocked when resizing from a corner'
+                    )
+                locked_aspect_ratio = True
+            else:
+                locked_aspect_ratio = False
+
+            locked_center = 'Control' in event.modifiers
+
+            # work in data space
+
+            if locked_center:
+                scaling_center = world_to_data(center)
```

Comment:
Can please you add a comment explaining what the "locked centre" and not "locked centre" behaviours are? It isn't clear to be at first glance.